src/js/views/OperationsLog/OperationsLog.jsx

```diff
@@ -1,25 +1,226 @@
+  const [searchParams, setSearchParams] = useSearchParams()
+  const [filter, setFilter] = useState(
+    searchParams.get('f') ? searchParams.get('f') : ''
+  )
+  const [onFetch, setOnFetch] = useState(true)
+  const [fetching, setFetching] = useState(false)
+  const [updated, setUpdated] = useState(false)
+  const [deletedID, setDeletedID] = useState()
+  const [rows, setRows] = useState([])
+  const [errorMessage, setErrorMessage] = useState()
+  const [showHelp, setShowHelp] = useState(false)
+  const { t } = useTranslation()
+
+  if (deletedID) {
+    setRows((prevRows) => prevRows.filter((r) => r.id !== deletedID))
+    setDeletedID(null)
+  }
+
   useEffect(() => {
     dispatch({
       type: 'SET_CURRENT_PAGE',
       payload: {
         title: 'operationsLog.title',
-        url: new URL('/ui/operations-log', state.baseURL)
+        url: new URL('/ui/operations-log', globalState.baseURL)
       }
     })
   }, [])
+
+  useEffect(() => {
+    if (fetching || !onFetch) return
+    setFetching(true)
+
+    const query = {
+      query: toElasticsearchQuery(
+        fromKueryExpression(filter),
+        metadataAsOptions.openSearch
+      ),
+      sort: [{ recorded_at: { order: 'desc' } }],
+      fields: globalState.fields,
+      size: 1000
+    }
+
+    httpPost(
+      globalState.fetch,
+      buildURL('/opensearch/operations-log'),
+      query
+    ).then(({ data, success }) => {
+      if (success) {
+        setErrorMessage(null)
+        setRows(
+          data.hits.hits
+            .map((hit) => hit._source)
+            .filter((r) => r.id !== deletedID)
+        )
+        setDeletedID(null)
+      } else {
+        setErrorMessage(t('projects.requestError', { error: data }))
+      }
+      setFetching(false)
+      setOnFetch(false)
+    })
+  }, [onFetch])
+
+  function buildURL(path) {
+    return new URL(path, globalState.baseURL)
+  }
+
+  const columns = [
+    {
+      title: t('operationsLog.changeType'),
+      name: 'change_type',
+      type: 'text',
+      tableOptions: {
+        headerClassName: 'w-2/12 truncate'
+      }
+    },
+    {
+      title: t('operationsLog.description'),
+      name: 'description',
+      type: 'text',
+      tableOptions: {
+        lookupFunction: (value) => <p className="truncate">{value}</p>
+      }
+    },
+    {
+      title: t('operationsLog.project'),
+      name: 'project_name',
+      type: 'text',
+      tableOptions: {
+        headerClassName: 'w-2/12',
+        lookupFunction: (value) => <p className="truncate">{value}</p>
+      }
+    },
+    {
+      title: t('operationsLog.environment'),
+      name: 'environment',
+      type: 'text',
+      tableOptions: {
+        headerClassName: 'w-2/12'
+      }
+    },
+    {
+      title: t('operationsLog.recordedBy'),
+      name: 'recorded_by',
+      type: 'text',
+      tableOptions: {
+        headerClassName: 'w-2/12 truncate',
+        lookupFunction: (value) => <p className="truncate">{value}</p>
+      }
```

Comment:
What do you think about adding a "datetime" `Column` type?  It could take care of the `displayValue` consistently (see below).

Reply:
After going back and forth on this, yeah I think it's fine to add that. I just had to clean up the Column logic a bit.